### ISSUE TITLE
NullPointerException with flatMap version that handles notification handling

### ISSUE THREAD

**Issue by mikegr** [ReactiveX/RxJava#3180]
The following code works with RxJava 1.0.12, but throws a NullPointerException with 1.0.13 and 1.0.14

```
Observable.just(1)
                .flatMap(
                        item -> Observable.just(item + 1),
                        e -> Observable.error(e),
                        () -> Observable.never())
                .subscribe(
                        i -> {
                            Log.d(TAG, "Got " + i);
                        },
                        e -> {
                            Log.d(TAG, "RxJava ERROR", e);
                        });

```

```
java.lang.NullPointerException: Attempt to invoke virtual method 'void rx.internal.operators.OperatorMapNotification$SingleEmitter.offer(java.lang.Object)' on a null object reference
at rx.internal.operators.OperatorMapNotification$1.onNext(OperatorMapNotification.java:82)
at rx.internal.util.ScalarSynchronousObservable$1.call(ScalarSynchronousObservable.java:46)
at rx.internal.util.ScalarSynchronousObservable$1.call(ScalarSynchronousObservable.java:35)
at rx.Observable$1.call(Observable.java:145)
at rx.Observable$1.call(Observable.java:137)
at rx.Observable$1.call(Observable.java:145)
at rx.Observable$1.call(Observable.java:137)
at rx.Observable.subscribe(Observable.java:7803)
at rx.Observable.subscribe(Observable.java:7579)
```

Further notice: Observable.just(1,2,3) works fine on all versions.

**Comment by akarnokd** [comment]
Thanks for reporting. #3181 fixes this issue.

**Comment by akarnokd** [comment]
Fix merged via #3181 and will be delivered in 1.0.15.
